python/test/t/registration/test_registration.py

```diff
@@ -0,0 +1,232 @@
+    assert convergence_criteria.relative_rmse == 1e-06
+
+
+@pytest.mark.parametrize("device", list_devices())
+def test_registration_result_constructor(device):
+    dtype = o3c.Dtype.Float32
+
+    #Constructor.
+    registration_result = o3d.t.pipelines.registration.RegistrationResult()
+
+    #Checking default values.
+    assert registration_result.inlier_rmse == 0.0
+    assert registration_result.fitness == 0.0
+    assert registration_result.transformation.allclose(
+        o3c.Tensor.eye(4, dtype, o3c.Device("CPU:0")))
+
+
+@pytest.mark.parametrize("device", list_devices())
+def test_evaluate_registration(device):
+    dtype = o3c.Dtype.Float32
+
+    source_points = o3c.Tensor(
+        [[1.15495, 2.40671, 1.15061], [1.81481, 2.06281, 1.71927],
+         [0.888322, 2.05068, 2.04879], [3.78842, 1.70788, 1.30246],
+         [1.8437, 2.22894, 0.986237], [2.95706, 2.2018, 0.987878],
+         [1.72644, 1.24356, 1.93486], [0.922024, 1.14872, 2.34317],
+         [3.70293, 1.85134, 1.15357], [3.06505, 1.30386, 1.55279],
+         [0.634826, 1.04995, 2.47046], [1.40107, 1.37469, 1.09687],
+         [2.93002, 1.96242, 1.48532], [3.74384, 1.30258, 1.30244]], dtype,
+        device)
+
+    target_points = o3c.Tensor(
+        [[2.41766, 2.05397, 1.74994], [1.37848, 2.19793, 1.66553],
+         [2.24325, 2.27183, 1.33708], [3.09898, 1.98482, 1.77401],
+         [1.81615, 1.48337, 1.49697], [3.01758, 2.20312, 1.51502],
+         [2.38836, 1.39096, 1.74914], [1.30911, 1.4252, 1.37429],
+         [3.16847, 1.39194, 1.90959], [1.59412, 1.53304, 1.5804],
+         [1.34342, 2.19027, 1.30075]], dtype, device)
+
+    target_normals = o3c.Tensor(
+        [[-0.0085016, -0.22355, -0.519574], [0.257463, -0.0738755, -0.698319],
+         [0.0574301, -0.484248, -0.409929], [-0.0123503, -0.230172, -0.52072],
+         [0.355904, -0.142007, -0.720467], [0.0674038, -0.418757, -0.458602],
+         [0.226091, 0.258253, -0.874024], [0.43979, 0.122441, -0.574998],
+         [0.109144, 0.180992, -0.762368], [0.273325, 0.292013, -0.903111],
+         [0.385407, -0.212348, -0.277818]], dtype, device)
+
+    source_t = o3d.t.geometry.PointCloud(device)
+    target_t = o3d.t.geometry.PointCloud(device)
+
+    source_t.point["points"] = source_points
+    target_t.point["points"] = target_points
+    target_t.point["normals"] = target_normals
+
+    source_legacy = source_t.to_legacy_pointcloud()
+    target_legacy = target_t.to_legacy_pointcloud()
+
+    max_correspondence_distance = 1.25
+    init_trans_legacy = np.eye(4)
+    init_trans_t = o3c.Tensor.eye(4, dtype, device)
+
+    evaluation_t = o3d.t.pipelines.registration.evaluate_registration(
+        source_t, target_t, max_correspondence_distance, init_trans_t)
+    evaluation_legacy = o3d.pipelines.registration.evaluate_registration(
+        source_legacy, target_legacy, max_correspondence_distance,
+        init_trans_legacy)
+
+    np.testing.assert_almost_equal(evaluation_t.inlier_rmse,
```

Comment:
What about `assert_allclose` (we typically use that in C++ unit tests as well)? What is the consideration between using `assert_almost_equal` v.s. `assert_allclose`?

Reply:
`assert_allclose` works! 
I found `assert_almost_equal` for single float / double value comparision, and `assert_allclose` for comparing general objects, but `assert_allclose` works. Done.